bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -84,18 +98,29 @@ public void run() {
         }
     };
 
-    protected void activate() {
```

Comment:
Question: If no activate() method is defined, is it guaranteed that modified() is called whenever a configuration (at activation time) is available? I usually also accept the config in activate and then call modified() myself - but maybe this isn't required at all?